infra/aws/main.tf

```diff
@@ -0,0 +1,163 @@
+      InstanceOperatingSystem = "linux"
+      OnDemandPercentage      = 100
+      EnableInstanceStorage   = "true"
+    },
+
+    gpu-1-queue = {
+      BuildkiteQueue          = "gpu_1_queue"
+      InstanceTypes           = "g6.4xlarge"
+      MaxSize                 = 60
+      ECRAccessPolicy         = "readonly"
+      InstanceOperatingSystem = "linux"
+      OnDemandPercentage      = 100
+      ImageId                 = "ami-060f16e7ab30bfef6"
+    },
+
+    gpu-4-queue = {
+      BuildkiteQueue          = "gpu_4_queue"
+      InstanceTypes           = "g6.12xlarge"
+      MaxSize                 = 2
+      ECRAccessPolicy         = "readonly"
+      InstanceOperatingSystem = "linux"
+      OnDemandPercentage      = 100
+      ImageId                 = "ami-060f16e7ab30bfef6"
+    }
+  }
+
+  merged_parameters = {
+    for name, params in local.queues_parameters :
+    name => merge(local.default_parameters, params)
+  }
+}
+
+
+resource "aws_cloudformation_stack" "bk_queue" {
+  for_each   = local.merged_parameters
+  name       = "bk-${each.key}"
+  parameters = { for k, v in each.value : k => v if k != "elastic_ci_stack_version" }
+
+  template_url = "https://s3.amazonaws.com/buildkite-aws-stack/v${each.value["elastic_ci_stack_version"]}/aws-stack.yml"
+  capabilities = ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM", "CAPABILITY_AUTO_EXPAND"]
+
+  lifecycle {
+    ignore_changes = [
+      tags["AppManagerCFNStackKey"],
+      tags_all["AppManagerCFNStackKey"],
+    ]
+  }
+}
+
+resource "aws_iam_policy" "ecr_public_access_policy" {
+  name        = "ecr-public-access-policy"
+  description = "Policy to push and pull images from ECR"
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "ecr-public:DescribeImageTags",
+          "ecr-public:DescribeRegistries",
+          "ecr-public:DescribeRepositories",
+          "ecr-public:BatchCheckLayerAvailability",
+          "ecr-public:DescribeImages",
+          "ecr-public:GetAuthorizationToken",
+          "ecr-public:GetRegistryCatalogData",
+          "ecr-public:GetRepositoryCatalogData",
+          "ecr-public:GetRepositoryPolicy",
+          "ecr-public:ListTagsForResource",
+          "ecr-public:CompleteLayerUpload",
+          "ecr-public:InitiateLayerUpload",
+          "ecr-public:PutImage",
+          "ecr-public:PutRegistryCatalogData",
+          "ecr-public:UploadLayerPart",
+          "ecr-public:TagResource",
+          "sts:GetServiceBearerToken"
+        ],
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+
+resource "aws_iam_policy" "bk_stack_secrets_access" {
+  name = "access-to-bk-stack-secrets"
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action = ["secretsmanager:GetSecretValue"],
+      Effect : "Allow",
+      Resource = [
+        "arn:aws:secretsmanager:us-west-2:936637512419:secret:hf_token-JD4TSm"
```

Comment:
shall we also declare the secret in terraform? so that this can be directly reference the output of the secret?

Reply:
?

Reply:
I added the secrets in terraform and change this to reference to that secret ARN